client/src/app/interceptors/auth.interceptor.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor,
+  HttpClient
+} from '@angular/common/http';
+import { environment } from 'src/environments/environment';
+import { catchError, Observable, throwError } from 'rxjs';
+import { User } from '../models/user.model'
+import { Token } from '../interfaces/token';
+
+
+/**
+ * Interceptor for HTTP requests, it attaches Authorization 
+ * Header and attempts to do a token refresh when  current 
+ * access Token expires. 
+ * At other layers ensure user is redirected to login 
+ * if Token expired error is thrown.
+ */
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  private  MAXRETRIES = 2;
+  private retries = 0;
+
+
+  constructor(private http: HttpClient) { }
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+
+    let accessToken = '';
+    const localUser = localStorage.getItem('user');
```

Comment:
Let's use the Auth service here (inject `AuthService` and use `auth.user`, which is a getter function)